### ISSUE TITLE
Aiyabot on docker can't start

### ISSUE THREAD

**Issue by HeartBtz** [Kilvoctu/aiyabot#186]
**Describe the bug**  

When starting the docker image with docker run 
`docker run --name KuroAI --network=host -e TOKEN=mytokendiscordblablabla -e URL="http://192.168.1.108:7860" -v /home/KuroAI/outputs:/app/outputs -v /home/KuroAI/resources:/app/resources -d ghcr.io/kilvoctu/aiyabot:latest`

---

**Environment**
 - OS: Linux debian 11 -> docker 

---

**Console logs**  
```
root@KuroAI:/home/KuroAI# docker logs  KuroAI
Using URL: http://192.168.1.108:7860
Using outputs directory: outputs
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/discord/cog.py", line 786, in _load_from_module_spec
    setup(self)
  File "/app/core/generatecog.py", line 76, in setup
    bot.add_cog(GenerateCog(bot))
                ^^^^^^^^^^^^^^^^
  File "/app/core/generatecog.py", line 19, in __init__
    tokenizer = GPT2Tokenizer.from_pretrained(self.model_path)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/transformers/tokenization_utils_base.py", line 1771, in from_pretrained
    resolved_vocab_files[file_id] = cached_file(
                                    ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/transformers/utils/hub.py", line 417, in cached_file
    resolved_file = hf_hub_download(
                    ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huggingface_hub/utils/_validators.py", line 110, in _inner_fn
    validate_repo_id(arg_value)
  File "/usr/local/lib/python3.11/site-packages/huggingface_hub/utils/_validators.py", line 158, in validate_repo_id
    raise HFValidationError(
huggingface_hub.utils._validators.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': 'core/MagicPrompt-SD/'. Use `repo_type` argument if needed.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/aiya.py", line 29, in <module>
    self.load_extension('core.generatecog')
  File "/usr/local/lib/python3.11/site-packages/discord/cog.py", line 910, in load_extension
    self._load_from_module_spec(spec, name)
  File "/usr/local/lib/python3.11/site-packages/discord/cog.py", line 791, in _load_from_module_spec
    raise errors.ExtensionFailed(key, e) from e
discord.errors.ExtensionFailed: Extension 'core.generatecog' raised an error: HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': 'core/MagicPrompt-SD/'. Use `repo_type` argument if needed.
root@KuroAI:/home/KuroAI#
```

Any idea of why ? :)

Have a nice day.

**Comment by solareon** [comment]
The issue is related to this code I believe. Try and add an extra volume mount for /app/core/MagicPrompt-SD and see if that clears it.

https://github.com/Kilvoctu/aiyabot/blob/5c35396adf78fe2598aa0dfcc3123db2a7fcaa97/core/setup_generate.py#L4-L5

**Comment by HeartBtz** [comment]
Hello :)

Thansk for your answer.

There is the result after adding new volume: 
` docker run --name KuroAI --network=host --restart=always -e TOKEN=`BoopdawopDiscordToken -e URL="http://192.168.1.108:7860" -v /home/KuroAI/outputs:/app/outputs -v /home/KuroAI/resources:/app/resources -v /home/KuroAI/MagicPrompt-SD:/app/core/MagicPrompt-SD -d ghcr.io/kilvoctu/aiyabot:latest`

```
root@KuroAI:~# docker logs KuroAI
Using URL: http://192.168.1.108:7860
Using outputs directory: outputs
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/discord/cog.py", line 786, in _load_from_module_spec
    setup(self)
  File "/app/core/generatecog.py", line 76, in setup
    bot.add_cog(GenerateCog(bot))
                ^^^^^^^^^^^^^^^^
  File "/app/core/generatecog.py", line 19, in __init__
    tokenizer = GPT2Tokenizer.from_pretrained(self.model_path)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/transformers/tokenization_utils_base.py", line 1796, in from_pretrained
    raise EnvironmentError(
OSError: Can't load tokenizer for 'core/MagicPrompt-SD/'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure 'core/MagicPrompt-SD/' is the correct path to a directory containing all relevant files for a GPT2Tokenizer tokenizer.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/aiya.py", line 29, in <module>
    self.load_extension('core.generatecog')
  File "/usr/local/lib/python3.11/site-packages/discord/cog.py", line 910, in load_extension
    self._load_from_module_spec(spec, name)
  File "/usr/local/lib/python3.11/site-packages/discord/cog.py", line 791, in _load_from_module_spec
    raise errors.ExtensionFailed(key, e) from e
discord.errors.ExtensionFailed: Extension 'core.generatecog' raised an error: OSError: Can't load tokenizer for 'core/MagicPrompt-SD/'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure 'core/MagicPrompt-SD/' is the correct path to a directory containing all relevant files for a GPT2Tokenizer tokenizer.
Using URL: http://192.168.1.108:7860
Using outputs directory: outputs
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/discord/cog.py", line 786, in _load_from_module_spec
    setup(self)
  File "/app/core/generatecog.py", line 76, in setup
    bot.add_cog(GenerateCog(bot))
                ^^^^^^^^^^^^^^^^
  File "/app/core/generatecog.py", line 19, in __init__
    tokenizer = GPT2Tokenizer.from_pretrained(self.model_path)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/transformers/tokenization_utils_base.py", line 1796, in from_pretrained
    raise EnvironmentError(
OSError: Can't load tokenizer for 'core/MagicPrompt-SD/'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure 'core/MagicPrompt-SD/' is the correct path to a directory containing all relevant files for a GPT2Tokenizer tokenizer.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/aiya.py", line 29, in <module>
    self.load_extension('core.generatecog')
  File "/usr/local/lib/python3.11/site-packages/discord/cog.py", line 910, in load_extension
    self._load_from_module_spec(spec, name)
  File "/usr/local/lib/python3.11/site-packages/discord/cog.py", line 791, in _load_from_module_spec
    raise errors.ExtensionFailed(key, e) from e
discord.errors.ExtensionFailed: Extension 'core.generatecog' raised an error: OSError: Can't load tokenizer for 'core/MagicPrompt-SD/'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure 'core/MagicPrompt-SD/' is the correct path to a directory containing all relevant files for a GPT2Tokenizer tokenizer.
root@KuroAI:~#
```

It seems it's not working :x

**Comment by solareon** [comment]
No worries. I appreciate the test and I'll look into it shortly when I get back to my computer

**Comment by HeartBtz** [comment]
I have a reason to use docker instead of normal clone :D
I have a proxmox server at home. 
At the moment, my 'stable diffusion' is on a LXC container. 
I'm trying to mount the bot on another LXC container. 
When I tried to clone the bot, it started downloading cuda and pytorch, and since I've done GPU passthrough on this container, it scared me and I preferred to switch back to docker, which worked the last time I tried. 
I obviously panicked when I saw that it was downloading cuda & pytorch.

I'll try again with your new docker image (on my lxc container :D), and if it doesn't work, I'll try again by cloning.

Translated with www.DeepL.com/Translator (free version)
